### PR TITLE
Add security center policy resources

### PR DESCRIPTION
**Pull request by omerdemirok**
Signed-off-by: Omer Demirok <odemirok@chef.io>

### Description

ADD
- Add `azure_security_center_policy` and `azure_security_center_policies` resources

UPDATE
-  Make `validate_resource_uri` method accept parameters and return result to be able to use it in additional API calls
- Improve some error messages
- Make resource type comparison case insensitive in `get_api_version` method

### Issues Resolved

N/A

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/azure_security_center_policies.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
+title: About the azure_security_center_policies Resource
+platform: azure
+---
+
+# azure_security_center_policies
+
+Use the `azure_security_center_policies` InSpec audit resource to test properties and configuration of multiple Azure Policys.
```
Comment (rmoles) [inline docs/resources/azure_security_center_policies.md:8]: Typo `multiple Azure Policys` --> `multiple Azure Policies`

File: libraries/azure_backend.rb [hunk 2]
```diff
@@ -313,14 +313,14 @@ def get_api_version(provider, resource_type, api_version_status = 'latest')
       provider_details = @azure.provider_details[provider.to_sym]
     end
 
-    resource_type_details = provider_details[:resourceTypes].select { |rt| rt[:resourceType] == resource_type }&.first
+    resource_type_details = provider_details[:resourceTypes].select { |rt| rt[:resourceType].downcase == resource_type.downcase }&.first
```
Comment (rmoles) [inline libraries/azure_backend.rb:316]: Slight nitpick, but I think in most cases we usually go with `upcase` for comparison

Comment (omerdemirok) [reply libraries/azure_backend.rb:316]: Will change the other comparisons as well.

File: docs/resources/azure_security_center_policies.md (hunk 1 above)
Comment (omerdemirok) [reply docs/resources/azure_security_center_policies.md]: This is fixed now.

**Pull request merged**